src/common/JsonFileManager.ts

```diff
@@ -31,9 +31,9 @@ export default class JsonFileManager<T> {
         });
     }
 
-    setJson(json: T): void {
+    async setJson(json: T): Promise<void> {
         this.json = json;
-        this.writeToFile();
+        await this.writeToFile();
```

Comment:
Just to make sure, did we test other uses of this function? I don't assume it'll break anything but worth having a quick look.